Classes/Line/JBLineChartView.m

```diff
@@ -1035,8 +1035,8 @@ - (NSInteger)lineIndexForPoint:(CGPoint)point
 				// Touch point
 				CGPoint normalizedTouchPoint = CGPointMake(point.x, self.linesView.bounds.size.height - point.y);
 				
-				// Slope
-				CGFloat lineSlope = (CGFloat)(rightPoint.y - leftPoint.y) / (CGFloat)(rightPoint.x - leftPoint.x);
+				// Slope - set to zero if x coordinates are the same and would result in a NaN value
```

Comment:
I usually put ternary conditions in the order of most desirable. Also 0.0f instead of 0:

CGFloat lineSlope = rightPoint.x != leftPoint.x ? (CGFloat)(rightPoint.y - leftPoint.y) / (CGFloat)(rightPoint.x - leftPoint.x) : 0.0f;